src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,595 @@
+				if (!String.IsNullOrEmpty (value))
+					ProcessPropertySignature (type, value, properties);
+
+				value = GetAttribute (properties.Current, "name");
+				if (!String.IsNullOrEmpty (value))
+					ProcessPropertyName (type, value, properties);
+			}
+		}
+
+		void ProcessFieldSignature (TypeDefinition type, string signature, XPathNodeIterator iterator)
+		{
+			FieldDefinition field = GetField (type, signature);
+			if (field == null) {
+				_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not find field '{signature}' in type '{type.FullName}' specified in { _xmlDocumentLocation}", 2016));
+				return;
+			}
+			ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", string.Empty));
+			DynamicallyAccessedMemberTypes fieldAnnotation = GetMemberTypesForDynamicallyAccessedMemberAttribute (attributes, _context, _xmlDocumentLocation);
+			if (fieldAnnotation != DynamicallyAccessedMemberTypes.None)
+				_fields[field] = fieldAnnotation;
+		}
+
+		void ProcessFieldName (TypeDefinition type, string name, XPathNodeIterator iterator)
+		{
+			if (!type.HasFields)
+				return;
+
+			foreach (FieldDefinition field in type.Fields) {
+				if (field.Name == name) {
+					ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", string.Empty));
+					DynamicallyAccessedMemberTypes fieldAnnotation = GetMemberTypesForDynamicallyAccessedMemberAttribute (attributes, _context, _xmlDocumentLocation);
+					if (fieldAnnotation != DynamicallyAccessedMemberTypes.None)
+						_fields[field] = fieldAnnotation;
+				}
+			}
+		}
+
+		static FieldDefinition GetField (TypeDefinition type, string signature)
+		{
+			if (!type.HasFields)
+				return null;
+
+			foreach (FieldDefinition field in type.Fields)
+				if (signature == field.FieldType.FullName + " " + field.Name)
+					return field;
+
+			return null;
+		}
+
+		void ProcessMethodSignature (TypeDefinition type, string signature, XPathNodeIterator iterator)
+		{
+			MethodDefinition method = GetMethod (type, signature);
+			if (method == null) {
+				_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not find method '{signature}' in type '{type.FullName}' specified in '{_xmlDocumentLocation}'", 2009));
+				return;
+			}
+			ProcessMethodChildren (type, method, iterator);
+		}
+
+		void ProcessMethodChildren (TypeDefinition type, MethodDefinition method, XPathNodeIterator iterator)
+		{
+			ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", string.Empty));
+			ArrayBuilder<(string, ArrayBuilder<Attribute>)> parameterAnnotations = ProcessParameters (type,
+				method, iterator.Current.SelectChildren ("parameter", string.Empty));
+			ArrayBuilder<ArrayBuilder<Attribute>> returnParameterAnnotations = ProcessReturnParameters (type,
+				method, iterator.Current.SelectChildren ("return", string.Empty));
```

Comment:
Nit: In theory it should be an error to have more than one "return" parameter. But I guess it's OK to allow it.